### PR TITLE
scream: 3.4 -> 3.6

### PR DESCRIPTION
**Pull request by arcnmx**
###### Motivation for this change

All receiver binaries have since been merged into one, so renamed from scream-receivers.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/scream/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config
+, alsaSupport ? true, alsaLib ? null
+, pulseSupport ? false, libpulseaudio ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/scream/default.nix:3]:
```suggestion
, alsaSupport ? true, alsaLib
, pulseSupport ? false, libpulseaudio
```

File: pkgs/applications/audio/scream/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    # Programs exit with code 1 when testing help, so grep for a string
+    $out/bin/scream -h 2>&1 | grep -q Usage:
+  '';
+
+  meta = with lib; {
+    description = "Audio receiver for the Scream virtual network sound card";
+    homepage = "https://github.com/duncanthrax/scream";
+    license = licenses.mspl;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/scream/default.nix:42]: Add yourself as a maintainer.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by arcnmx** [comment]
Rebased onto master and made the suggested changes, sorry that took so long.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -8060,8 +8060,9 @@ in
 
   scmpuff = callPackage ../applications/version-management/git-and-tools/scmpuff { };
 
-  scream-receivers = callPackage ../misc/scream-receivers {
+  scream = callPackage ../applications/audio/scream {
     pulseSupport = config.pulseaudio or false;
+    alsaSupport = stdenv.isLinux;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:8065]:
```suggestion
```

File: pkgs/applications/audio/scream/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config
+, alsaSupport ? true, alsaLib
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/scream/default.nix:2]:
```suggestion
, alsaSupport ? stdenv.isLinux, alsaLib
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -8060,8 +8060,9 @@ in
 
   scmpuff = callPackage ../applications/version-management/git-and-tools/scmpuff { };
 
-  scream-receivers = callPackage ../misc/scream-receivers {
+  scream = callPackage ../applications/audio/scream {
     pulseSupport = config.pulseaudio or false;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:8064]:
```suggestion
```

File: pkgs/applications/audio/scream/default.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+, pulseSupport ? false, libpulseaudio
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/scream/default.nix:3]:
```suggestion
, pulseSupport ? config.pulseaudio or false, libpulseaudio
```

File: pkgs/applications/audio/scream/default.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  src = fetchFromGitHub {
+    owner = "duncanthrax";
+    repo = pname;
+    rev = version;
+    sha256 = "01k2zhfb781gfj3apmcjqbm5m05m6pvnh7fb5k81zwvqibai000v";
+  };
+
+  buildInputs = lib.optional pulseSupport libpulseaudio
+    ++ lib.optional alsaSupport alsaLib;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/scream/default.nix:18]:
```suggestion
  buildInputs = lib.optionals pulseSupport [ libpulseaudio ]
    ++ lib.optionals alsaSupport [ alsaLib ];
```

File: pkgs/applications/audio/scream/default.nix [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  cmakeFlags = [
+    "-DPULSEAUDIO_ENABLE=${if pulseSupport then "ON" else "OFF"}"
+    "-DALSA_ENABLE=${if alsaSupport then "ON" else "OFF"}"
+  ];
+
+  cmakeDir = "../Receivers/unix";
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    set -o verbose
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/scream/default.nix:30]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
The suggestions from me are more commonly used and make it easier to know when something is enabled and not.

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (arcnmx) [reply pkgs/top-level/all-packages.nix]: Looking at other packages, `config.pulseaudio or stdenv.isLinux` is more common and sensical, so changed to that.

File: pkgs/applications/audio/scream/default.nix (hunk 6 above)
Comment (arcnmx) [reply pkgs/applications/audio/scream/default.nix:3]: Referencing `config` inside the derivation feels a bit odd, and other packages seem to set it in all-packages.nix, so not sure about doing this?

File: pkgs/applications/audio/scream/default.nix (hunk 7 above)
Comment (arcnmx) [reply pkgs/applications/audio/scream/default.nix:18]: This one confuses me, isn't this the exact scenario that `optional` is for (as shown in its example docs)? is `optional` considered bad form or deprecated or something?

File: pkgs/applications/audio/scream/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/scream/default.nix:3]: Looking at more packages define it in the package itself: https://search.nix.gsc.io/?q=config.pulseaudio&i=nope&files=&repos=

It makes it clear when the option changes without changing files.

File: pkgs/applications/audio/scream/default.nix (hunk 7 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/scream/default.nix:18]: optional accepts any datatype. optionals only accepts lists and is supposed to be used in conjunction with lists.

Comment (arcnmx) [reply pkgs/applications/audio/scream/default.nix:18]: *points to docs and "flashplayer" example: [lib.lists.optional](https://nixos.org/manual/nixpkgs/stable/#function-library-lib.lists.optional)*

`optional a b` is like a singleton version of optionals, it's equivalent to `optional a (singleton b)` or more simply `optionals a [ b ]`. This is what it's for afaik, when there's only a single item to be added to the list.

**Comment by arcnmx** [comment]
@SuperSandro2000 ping?

**Pull request merged**